src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -674,87 +675,93 @@ private Index HandleStaticIndexChange(string name, IndexDefinition definition)
+                            if (mapReduceIndex.OutputReduceToCollection != null && prefixesOfDocumentsToDelete?.Count > 0)
+                                mapReduceIndex.OutputReduceToCollection.AddPrefixesOfDocumentsToDelete(prefixesOfDocumentsToDelete);
 
-                        break;
+                            index = mapReduceIndex;
+                            break;
+                        default:
+                            throw new NotSupportedException($"Cannot create {indexDefinition.Type} index from IndexDefinition");
+                    }
+                    break;
+                case IndexSourceType.Counters:
+                    switch (indexDefinition.Type)
+                    {
+                        case IndexType.Map:
+                        case IndexType.JavaScriptMap:
+                            index = MapCountersIndex.CreateNew(indexDefinition, documentDatabase, configuration: optionalIndexConfiguration);
+                            break;
+                        case IndexType.MapReduce:
+                        case IndexType.JavaScriptMapReduce:
+                            var mapReduceIndex = MapReduceIndex.CreateNew<MapReduceCountersIndex>(indexDefinition, documentDatabase, configuration: optionalIndexConfiguration);
+                            if (mapReduceIndex.OutputReduceToCollection != null && prefixesOfDocumentsToDelete?.Count > 0)
+                                mapReduceIndex.OutputReduceToCollection.AddPrefixesOfDocumentsToDelete(prefixesOfDocumentsToDelete);
 
-                    default:
-                        throw new NotSupportedException($"Not supported source type '{definition.SourceType}'.");
-                }
+                            index = mapReduceIndex;
+                            break;
+                        default:
+                            throw new NotSupportedException($"Cannot create {indexDefinition.Type} index from IndexDefinition");
+                    }
+                    break;
+                case IndexSourceType.TimeSeries:
+                    switch (indexDefinition.Type)
+                    {
+                        case IndexType.Map:
+                        case IndexType.JavaScriptMap:
+                            index = MapTimeSeriesIndex.CreateNew(indexDefinition, documentDatabase, optionalIndexConfiguration);
+                            break;
+                        case IndexType.MapReduce:
+                        case IndexType.JavaScriptMapReduce:
+                            var mapReduceIndex = MapReduceIndex.CreateNew<MapReduceTimeSeriesIndex>(indexDefinition, documentDatabase, configuration: optionalIndexConfiguration);
+                            
+                            if (mapReduceIndex.OutputReduceToCollection != null && prefixesOfDocumentsToDelete?.Count > 0)
+                                mapReduceIndex.OutputReduceToCollection.AddPrefixesOfDocumentsToDelete(prefixesOfDocumentsToDelete);
```

Comment:
1. This piece of code is repeated 3 times here. Can we do it just once?
2. Maybe instead of passing `prefixesOfDocumentsToDelete` here we could handled that in `HandleStaticIndexChange()`

Reply:
I moved it to `HandleStaticIndexChange`